bluesky-social/social-app
# Fix drawer swipe

See main insight in https://github.com/software-mansion/react-native-gesture-handler/issues/3266#issuecomment-2525301903.

New approach relies on https://github.com/react-navigation/react-navigation/pull/12326 which I expect to get merged soon.

## Test Plan

- Try swiping between tabs on Home and Profile
  - Verify the drawer gesture only activates on the first page, and that the pager takes over on other pages
  - Verify the drawer gesture doesn't steal vertical scrolling
  - Try to switch pages fast, in different ways, spam the left/right gesture
    - Note that it won't immediately open the drawer if you just arrived to the first tab without resting. That's kind of intentional to disambiguate pager swipes and intentional drawer swipe. Maybe fixable but let's not fix that yet.
- Switch between Home and Profile, or Home and Notifications
  - Verify only the current screen's selection determines whether the gesture is enabled
    - For example, open second tab on Home but first tab on Profile. Second tab on Home should disable it (because it's second) but *only while you're on Home*. First tab on Profile should enable it (because it's first), but *only while you're on Profile*. Switching between Home and Profile should update the gesture activeness just like switching between pager tabs on those respective root tabs.
  - Verify that the gesture never activates if the nav stack has something on it
  - Verify root screens without tabs (notifications, chat) always have the gesture enabled
